README.md

```diff
@@ -2,31 +2,173 @@
-referencing.
+1. A list is an ordered data structure, which means that each element is added
+   at a specific location and can be accessed using an index, which represents
+   its position in the collection.
+2. Lists allow duplicate values.
+3. A list is dynamically sized, which means it can continue to grow as you add
+   elements to it.
+4. A list is part of the Java Collections framework, which gives it several
+   capabilities that we will go over below.
 
-## Conclusion
+In this section, we will focus on general `List` functionality, and we will use
+the most commonly used `List` implementation, `ArrayList`, in our examples. But
+all the examples we will be going through will work with any of the `List`
+implementations.
 
-A short one or two paragraph summary of the contents of the lessons, recapping
-the learning goals.
+## ArrayList in Action
 
-## Resources
+To use a `List` in Java, we must choose a specific implementation - in this case,
+we will choose an ArrayList. An **ArrayList** is a class that implements the
+`List` interface and supports the addition and removal of elements, unlike an
+array. To make use of the `List` data structure, we need to import it. The
+`List` interface and the `ArrayList` class both live in the `java.util` package:
 
-- [Resource Link 1](example.com)
-- [Resource Link 2](example.com)
+```java
+import java.util.List;
+import java.util.ArrayList;
+
+public class ListExample {
+    public static void main(String[] args) {
+       List myList = new ArrayList();
+       myList.add("a name");
+       myList.add(2);
+       myList.add(2.3);
+       for (Object item: myList) {
+          System.out.println(item);
+       }
+       myList.add("a fourth item");
+       System.out.println(myList.get(3));
+    }
+}
+```
+
+As we can see, we were able to create a list without specifying how many items
+it would hold, and we were able to keep adding to the list without having to worry
+about re-sizing it. This differs from the arrays that we learned about earlier
+since arrays are a fixed size.
+
+We can refer to any item on the list by its index: `myList.get(<index>);`. Note
+that, as with arrays, indexes on lists start with the number 0.
+
+Did you notice that we were able to add objects of any kind to this list? This is
+not necessarily a good thing:
+
+1. Although it's convenient to be able to put anything in the list, it makes
+   using any of those values difficult because we don't know their type. This
+   means we can't be sure what we're allowed to do with them and what we're not.
+2. This is not "type safe", which means the compiler has no way to tell if we're
+   adding something to the list that wasn't meant to be there.
+
+Let's specify what type of items we want in the list to make it type safe:
+
+```java
+import java.util.List;
+import java.util.ArrayList;
+
+public class ListExample {
+    public static void main(String[] args) {
+       List<String> myStringList = new ArrayList<String>();
```

Comment:
You have `List` here but in the examples below you have `ArrayList`.
Would it make sense to explain to students in this lesson why they should declare the type as a List rather than ArrayList?
Also, I don't think you need the String type parameter in both sets of angled brackets.